onnxruntime/core/framework/utils.h

```diff
@@ -156,7 +156,7 @@ constexpr ONNXTensorElementDataType GetONNXTensorElementDataType<uint64_t>() {
 
 int32_t ONNXTensorElementDataTypeToProtoTensorType(ONNXTensorElementDataType);
 
-#ifdef ENABLE_TRAINING
+#if defined(ENABLE_TRAINING) || defined(ENABLE_TRAINING_OPS)
```

Comment:
Ops that need Contiguous inputs are NcclAllReduce, NcclReduceScatter, NcclAllGather... 
I don't think it need defined(ENABLE_TRAINING_OPS) here.